spec.md

```diff
@@ -179,6 +218,49 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+### Public Modules
+
+A public module is a module that is marked with the `public`{.firrtl} keyword.
+A public module has a defined ABI---knowing the FIRRTL representation is
+sufficient to know its compiled representation.  This means that a public module
+is available external to this circuit.
+
+The example below shows a public module `Foo`:
+
+``` firrtl
+public module Foo:
+  input a: UInt<1>
+  output b: UInt<1>
+
+  connect b, a
+```
+
+A public module has a number of restrictions:
+
+1. A public module may have no ports of uninferred width.
+1. A public module may have no ports of abstract reset type.
+1. A public module may have no ports of input probe type.
+1. A `RWProbe`{.firrtl} may not be used to access a public module's ports.
+1. A public module's ports are lowered using one of the port lowering ABIs
+   defined in the FIRRTL ABI specification.
+1. A public module may be instantiated by other modules, but must be compiled
```

Comment:
Alternately: "A public module may be instantiated by other modules within a circuit, but the behavior of the module must not be affected by these instantiations."